src/components/Transition/Transition.tsx

```diff
@@ -0,0 +1,109 @@
+import * as PropTypes from 'prop-types'
+import * as React from 'react'
+
+import { UIComponent, customPropTypes, createShorthandFactory, childrenExist } from '../../lib'
+import { ComponentSlotStyle, ComponentVariablesInput, Animation } from '../../themes/types'
+import { ReactChildren } from '../../../types/utils'
+import createAnimationStyles from '../../lib/createAnimationStyles'
+
+export type TransitionProps = {
+  animationName?: string
+  as?: any
+  children?: ReactChildren
+  delay?: string
+  direction?: string
+  duration?: string
+  fillMode?: string
+  iterationCount?: string
+  keyframeParams?: object
+  playState?: string
+  styles?: ComponentSlotStyle
+  timingFunction?: string
+  variables?: ComponentVariablesInput
+}
+
+/**
+ * A transition is an animation usually used to move content in or out of view.
+ */
+class Transition extends UIComponent<TransitionProps, any> {
+  static create: Function
+
+  static className = 'ui-transition'
+
+  static displayName = 'Transition'
+
+  static propTypes = {
+    /** The animation to be run. */
+    animationName: PropTypes.string,
+
+    /** An element type to render as. */
+    as: customPropTypes.as,
+
+    /**
+     *  Button content for childrenApi
+     *  @docSiteIgnore
+     */
+    children: PropTypes.node,
+
+    /** Delay of the animation. */
```

Comment:
Let's remove these descriptions or update to what fela has.  Ideally, these would give the user more detail about why/how they affect the transition.